library/src/main/java/com/google/android/exoplayer/text/subrip/SubripParser.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2014 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer.text.subrip;
+
+import android.text.Html;
+
+import com.google.android.exoplayer.C;
+import com.google.android.exoplayer.ParserException;
+import com.google.android.exoplayer.text.Cue;
+import com.google.android.exoplayer.text.SubtitleParser;
+import com.google.android.exoplayer.util.MimeTypes;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A simple SRT parser.
+ * <p/>
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/SubRip">Wikipedia on SRT</a>
+ */
+public class SubripParser implements SubtitleParser {
+
+  private static final String TAG = "SubRipParser";
+
+  private static final String SUBRIP_POSITION_STRING = "^(\\d)$";
+  private static final Pattern SUBRIP_POSITION = Pattern.compile(SUBRIP_POSITION_STRING);
+
+  private static final String SUBRIP_CUE_IDENTIFIER_STRING = "^(.*)\\s-->\\s(.*)$";
+  private static final Pattern SUBRIP_CUE_IDENTIFIER =
+          Pattern.compile(SUBRIP_CUE_IDENTIFIER_STRING);
+
+  private static final String SUBRIP_TIMESTAMP_STRING = "(\\d+:)?[0-5]\\d:[0-5]\\d:[0-5]\\d,\\d{3}";
+  // private static final Pattern SUBRIP_TIMESTAMP = Pattern.compile(SUBRIP_TIMESTAMP_STRING);
+
+  private final StringBuilder textBuilder;
+
+  private final boolean strictParsing;
+
+  public SubripParser() {
+    this(true);
+  }
+
+  public SubripParser(boolean strictParsing) {
+    this.strictParsing = strictParsing;
+
+    textBuilder = new StringBuilder();
+  }
+
+  @Override
+  public SubripSubtitle parse(InputStream inputStream, String inputEncoding, long startTimeUs)
+          throws IOException {
+    ArrayList<SubripCue> subtitles = new ArrayList<>();
+
+    // file should not be empty
+    if (inputStream.available() == 0) {
```

Comment:
It's allowable for an InputStream to say it has nothing available when it actually has something available. IMO you can just not do this. It'll end up parsing an empty subtitle in this case, I think, which is fine?